src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java

```diff
@@ -0,0 +1,49 @@
+package org.fogbowcloud.saps.engine.core.catalog;
+
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.List;
+
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.CatalogException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.TaskNotFoundException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.UserNotFoundException;
+import org.fogbowcloud.saps.engine.core.dispatcher.notifier.Ward;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.core.model.SapsUser;
+import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
+
+public interface Catalog {
+
+    SapsImage addTask(String taskId, String dataset, String region, Date date, int priority, String user,
+                      String inputdownloadingPhaseTag, String digestInputdownloading, String preprocessingPhaseTag,
+                      String digestPreprocessing, String processingPhaseTag, String digestProcessing) throws CatalogException;
+
+    void addStateChangeTime(String taskId, ImageTaskState state, Timestamp timestamp) throws CatalogException;
+
+    void addUser(String userEmail, String userName, String userPass, boolean userState, boolean userNotify,
```

Comment:
addUser is also complex. what is an userState? didn't we decide to remove all Notify stuff? userName and userPass have a max size?

Reply:
Doc added.
`userState` is related to whether the user can authenticate to the SAPS platform, when a new user is created through the dashboard, it is set to false, requiring an admin to enable it.
`userNotify` is related to whether the user wants to receive notifications about their SAPS tasks, but at the moment there is no feature of that.
There is no maximum size for `userName` or `userPass` parameters